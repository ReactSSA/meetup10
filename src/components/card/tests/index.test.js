import React from 'react';
import Card from '../';
import { shallow } from 'enzyme';
import { card } from '../../../data';

describe('Card', () => {
  const onAdd = jest.fn();
  const onRemove = jest.fn();
  const renderShallow = () => shallow(
    <Card card={card} onAdd={onAdd} onRemove={onRemove} />
  );

  describe('render', () => {
    it('Deve renderizar corretamente', () => {
      const wrapper = renderShallow();
      expect(wrapper).toMatchSnapshot();
    });

    it('Deve renderizar corretamente aberto', () => {
      const wrapper = renderShallow();
      wrapper.setState({ isOpen: true });
      expect(wrapper).toMatchSnapshot();
    });

    it('Deve renderizar corretamente adicionado', () => {
      const wrapper = renderShallow();
      wrapper.setState({ isAdded: true });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    it('Deve setar o estado corretamente depois do clique', () => {
      const wrapper = renderShallow();
      expect(wrapper.state('isOpen')).toBeFalsy();
      const img = wrapper.find('.cardBody');
      img.simulate('click');
      expect(wrapper.state('isOpen')).toBeTruthy();
    });

    it('Deve setar o estado corretamente onClose', () => {
      const wrapper = renderShallow();
      wrapper.setState({ isOpen: true });
      const CardDetails = wrapper.find('CardDetails');
      CardDetails.simulate('close');
      expect(wrapper.state('isOpen')).toBeFalsy();
    });

    it('Deve remover uma carta no clique do botão', () => {
      const wrapper = renderShallow();
      wrapper.setState({ isAdded: true });
      const button = wrapper.find('.btn');
      button.simulate('click');
      expect(wrapper.state('isAdded')).toBeFalsy();
      expect(onRemove).toHaveBeenCalledWith(1);
    });

    it('Deve adicionar uma carta no clique do botão', () => {
      const wrapper = renderShallow();
      const button = wrapper.find('.btn');
      button.simulate('click');
      expect(wrapper.state('isAdded')).toBeTruthy();
      expect(onAdd).toHaveBeenCalledWith({
        name: card.name,
        cardId: card.id,
      });
    });
  });
});
