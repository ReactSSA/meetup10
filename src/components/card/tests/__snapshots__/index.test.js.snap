// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render Deve renderizar corretamente 1`] = `
<div
  className="cardWrapper"
>
  <img
    alt="imagem"
    className="cardBody"
    onClick={[Function]}
  />
  <div
    className="cardFooter"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      Adicionar
    </button>
  </div>
</div>
`;

exports[`Card render Deve renderizar corretamente aberto 1`] = `
<div
  className="cardWrapper"
>
  <img
    alt="imagem"
    className="cardBody"
    onClick={[Function]}
  />
  <CardDetails
    artist="murtinha"
    flavor="Card flavor"
    manaCost="0"
    name="Black Lotus"
    onClose={[Function]}
    rarity="Rare"
    setName="Magic collection"
    text="Card text"
    type="Artifact"
  />
  <div
    className="cardFooter"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      Adicionar
    </button>
  </div>
</div>
`;

exports[`Card render Deve renderizar corretamente adicionado 1`] = `
<div
  className="cardWrapper"
>
  <img
    alt="imagem"
    className="cardBody"
    onClick={[Function]}
  />
  <div
    className="cardFooter"
  >
    <button
      className="btn added"
      onClick={[Function]}
    >
      Remover
    </button>
  </div>
</div>
`;
